src/VisualStudio/Core/Def/Implementation/Experimentation/KeybindingResetDetector.cs

```diff
@@ -139,12 +142,22 @@ private void InitializeCore()
         }
 
         private void UpdateStateMachine()
+        {
+            ThreadHelper.JoinableTaskFactory.RunAsync(async delegate
+            {
+                await UpdateStateMachineAsync();
+            });
+        }
+
+        private async Task UpdateStateMachineAsync()
         {
             AssertIsForeground();
 
-            var currentStatus = IsReSharperEnabled();
             var options = _workspace.Options;
             ReSharperStatus lastStatus = options.GetOption(KeybindingResetOptions.ReSharperStatus);
+            var currentStatus = await IsReSharperRunningAsync(lastStatus).ConfigureAwait(false);
```

Comment:
also, if you are using JTF, you shouldn't use ConfigureAwait but let it use JTF context it have set when you used JTF.